### ISSUE TITLE
How to verify the response content in async tests?

### ISSUE THREAD

**Issue by moroshko** [DiUS/pact-consumer-js-dsl#39]
Looking at [this example](https://github.com/DiUS/pact-consumer-js-dsl/blob/master/example/web/client-spec.js#L83), I wonder what is the point of specifying the `body` in `.willRespondWith()`?

Looks like the [body is verified in the `success` function](https://github.com/DiUS/pact-consumer-js-dsl/blob/master/example/web/client-spec.js#L103), and if [this line](https://github.com/DiUS/pact-consumer-js-dsl/blob/master/example/web/client-spec.js#L103) is removed, the test passes (so the `body` that is specified in `.willRespondWith()` is basically ignored).

In other words, looks like I need to duplicate the expected response in 2 places: 
* `body` in `.willRespondWith()`
* `expect` in `success()`

Is there a way to avoid the duplication?

**Comment by bethesque** [comment]
The `.willRespondWith()` is required to be set to stub out the response, and to allow us verify (on the provider side) that the real response matches the actual response. If you took that out, the test would fail, because the client code would have no body to parse.

The expectation in `success` asserts that your code has correctly handled the response.

**Comment by moroshko** [comment]
Thanks, makes sense now
